### PR TITLE
Extend CUB DeviceSegmentedReduce API with fixed segment size to support all operators

### LINKED ISSUE #4474: [FEA]: Extend CUB DeviceSegmentedReduce API with fixed segment size to support all operators
### Is this a duplicate?

- [x] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cccl/issues) for this request and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Area

CUB

### Is your feature request related to a problem? Please describe.

#3969 introduced **DeviceSegmentedReduce::Reduce()** API supporting fixed segment size segmented reduction. But, it misses other API entry points such as **Sum(), Min(), Max(), Argmin() and ArgMax()**.

### Describe the solution you'd like

Add the missing APIs by dispatching to lower level dispatch layer with appropriate reduction operator. 

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by srinivasyadav18**
## Description

closes #4474

## Checklist

- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cub/cub/device/device_segmented_reduce.cuh [hunk 1]
```diff
@@ -751,6 +936,116 @@ public:
...
+  //!     :end-before: example-end fixed-size-segmented-reduce-argmin
+  //!
+  //! @endrst
+  //!
+  //! @tparam InputIteratorT
+  //!   **[inferred]** Random-access input iterator type for reading input items (of some type `T`) @iterator
+  //!
+  //! @tparam OutputIteratorT
+  //!   **[inferred]** Output iterator type for recording the reduced aggregate
+  //!   (having value type `KeyValuePair<int, T>`) @iterator
```
Comment (elstehle) [inline cub/cub/device/device_segmented_reduce.cuh:968]: important:
```suggestion
  //!   (having value type `cuda::std::pair<int, T>`) @iterator
```

File: cub/cub/device/device_segmented_reduce.cuh [hunk 2]
```diff
@@ -49,13 +49,41 @@
...
+// @brief Functor to generate a key-value pair from an index and value
+template <typename Iterator, typename OutputValueT>
+struct generate_idx_value
+{
+private:
+  Iterator it;
+  int segment_size;
+
+public:
+  __host__ __device__ generate_idx_value(Iterator it, int segment_size)
```
Comment (elstehle) [inline cub/cub/device/device_segmented_reduce.cuh:75]:
```suggestion
  _CCCL_HOST_DEVICE _CCCL_FORCEINLINE generate_idx_value(Iterator it, int segment_size)
```

File: cub/cub/device/device_segmented_reduce.cuh [hunk 3]
```diff
@@ -49,13 +49,41 @@
...
+      : it(it)
+      , segment_size(segment_size)
+  {}
+
+  __host__ __device__ auto operator()(::cuda::std::int64_t idx) const
```
Comment (elstehle) [inline cub/cub/device/device_segmented_reduce.cuh:80]:
```suggestion
  _CCCL_HOST_DEVICE _CCCL_FORCEINLINE auto operator()(::cuda::std::int64_t idx) const
```

**Review by elstehle — APPROVED** [review]
Couple minor comments, otherwise looks good. Thanks for the great work!

File: c2h/include/c2h/custom_type.h [hunk 4]
```diff
@@ -69,7 +69,7 @@ class less_comparable_t
 {
   // The CUDA compiler follows the IA64 ABI for class layout, while the
   // Microsoft host compiler does not.
-  char workaround_msvc;
+  char workaround_msvc{};
```
[the remark this answers is not in the document]
Comment (srinivasyadav18) [reply c2h/include/c2h/custom_type.h:72]: The **compiler complained** that `workaround_msvc` might be **used uninitialized**. So, to fix that we just **zero initialized** it.

File: cub/test/catch2_test_device_segmented_reduce.cu [hunk 5]
```diff
@@ -310,4 +286,110 @@ C2H_TEST("Device fixed size segmented reduce works with all device interfaces",
...
+    REQUIRE(h_expected_result == h_out_result);
+  }
+
+  SECTION("argmin")
+  {
+    using result_t = ::cuda::std::pair<int, output_t>;
+
+    // Prepare verification data
+    c2h::host_vector<result_t> h_expected_result(num_segments);
+    compute_fixed_size_segmented_argmin_reference(in_items, num_segments, segment_size, h_expected_result.begin());
```
Comment (elstehle) [inline cub/test/catch2_test_device_segmented_reduce.cu:344]: Could you make sure that we also test for empty segments (i.e., `segment_size=0`). I think you can just add this to the `GENERATE_COPY` above:

```
const segment_size_t segment_size = GENERATE_COPY(
values({0}),
take(2, random(1 << 0, 1 << 5)),
...
```

File: cub/test/catch2_test_device_segmented_reduce_large_offsets.cu [hunk 6]
```diff
@@ -276,71 +300,142 @@ struct dispatch_helper
   }
 };
 
+// generic test for fixed size segmented reduce
+template <bool IS_REDUCE_ALGORITHM,
```
Comment (elstehle) [inline cub/test/catch2_test_device_segmented_reduce_large_offsets.cu:304]: We're generally using `CamelCase` for template parameters.
```suggestion
template <bool IsReduceAlgorithm,
```

**Review by elstehle — COMMENTED** [review]
Adding few minor comments after latest changes.
